### ISSUE TITLE
Playground: strictOptionalProperties not working on nightly TS version

### ISSUE THREAD

**Issue by josephjunker** [microsoft/TypeScript-Website#1912]
Apologies if reporting this issue is premature, given how recently the 4.4 beta build was released. I wanted to test out the new `strictOptionalProperties` setting in the playground, but there's no option for this in the `TSConfig` menu under the "Strict checks" section. If I enable the "strict" setting, the old optional properties behavior is still used.

<img width="1130" alt="Playground nightly TSConfig" src="https://user-images.githubusercontent.com/1519098/124019074-728f0300-d99d-11eb-883d-b0732dcad330.png">
<img width="1115" alt="Playground nightly sample code" src="https://user-images.githubusercontent.com/1519098/124019117-7fabf200-d99d-11eb-8fde-607645314dca.png">

**Repro:**
- Set the playground to use the nightly build, and enable strict mode.
- Enter this code:
```typescript
type foo = Partial<{bar: number}>
const test: foo = { bar: undefined }
```

**Expected behavior:**
A type error should be reported. Additionally, intellisense when mousing-over `foo` should display
```typescript
type foo = {
    bar?: number;
}
```

**Actual behavior:**
No type error is reported. Intellisense when mousing over `foo` displays
```typescript
type foo = {
    bar?: number | undefined;
}
```

**Comment by orta** [comment]
It's OK, the compiler flag was renamed to `exactOptionalPropertyTypes` and taken out of strict

https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.0-dev.20210630#code/C4TwDgpgBAZg9nKBeKBvAUFLUBGBDAJwH4AuKAOwFcBbHCAgbnQF910g
